File: Intermediate/Topic-005-Array-Carry-forward-&-Subarrays/Assigment/Q3-Special-Subsequences-AG/Special-Subsequences-AG.c
/**
 * @input A : String termination by '\0'
 * 
 * @Output Long
 */
long long solve(char* A) {
    long long count = 0;
    long long count_a = 0;
    int i;
    for (i = 0; A[i] != '\0'; i++) {
        if (A[i] == 'A') {
            count_a += 1;
        }

        if (A[i] == 'G') {
            count += count_a;
        }
    }

    return count;
}

File: Intermediate/Topic-005-Array-Carry-forward-&-Subarrays/Assigment/Q3-Special-Subsequences-AG/Special-Subsequences-AG.py
class Solution:
    # @param A : string
     # @return an long
    def solve(self, A):
        count = 0
        count_a = 0

        for i in range(len(A)):
            if (A[i] == 'A'):
                count_a += 1
            if (A[i] == 'G'):
                count += count_a
        return count
